app/src/main/java/ru/woodymsk/socialapp/presentation/activity/MainActivity.kt

```diff
@@ -1,79 +1,133 @@
 import ru.woodymsk.socialapp.R.id.itBottomNavigationAuthScreen
 import ru.woodymsk.socialapp.R.id.itBottomNavigationEventScreen
 import ru.woodymsk.socialapp.R.id.itBottomNavigationMyProfileScreen
 import ru.woodymsk.socialapp.R.id.itBottomNavigationPostScreen
 import ru.woodymsk.socialapp.data.auth.AppAuth
 import ru.woodymsk.socialapp.databinding.ActivityMainBinding
-import ru.woodymsk.socialapp.domain.Navigator
-import ru.woodymsk.socialapp.presentation.auth.AuthFragment
-import ru.woodymsk.socialapp.presentation.event.EventScreenFragment
-import ru.woodymsk.socialapp.presentation.my_profile.MyProfileScreenFragment
-import ru.woodymsk.socialapp.presentation.post.PostScreenFragment
+import ru.woodymsk.socialapp.domain.navigation.BottomNavigation
+import ru.woodymsk.socialapp.domain.navigation.subnavigation.LocalCiceroneHolder
+import ru.woodymsk.socialapp.presentation.activity.model.CurrentTabState
+import ru.woodymsk.socialapp.presentation.auth.AuthTabContainerFragment
+import ru.woodymsk.socialapp.presentation.common.BackButtonListener
+import ru.woodymsk.socialapp.presentation.common.TabTag
+import ru.woodymsk.socialapp.presentation.event.EventTabContainerFragment
+import ru.woodymsk.socialapp.presentation.my_profile.MyProfileContainerFragment
+import ru.woodymsk.socialapp.presentation.post.PostTabContainerFragment
 import javax.inject.Inject
 
 @AndroidEntryPoint
-class MainActivity : AppCompatActivity(), Navigator {
+class MainActivity : AppCompatActivity(), BottomNavigation, BackButtonListener {
+
+    private val viewModel: MainViewModel by viewModels()
+    private val navigator: com.github.terrakok.cicerone.Navigator =
+        object : AppNavigator(this, fragmentContainer) {
+            override fun applyCommands(commands: Array<out Command>) {
+                super.applyCommands(commands)
+                supportFragmentManager.executePendingTransactions()
+            }
+        }
+
+    @Inject
+    lateinit var ciceroneHolder: LocalCiceroneHolder
 
     @Inject
     lateinit var auth: AppAuth
 
+    @Inject
+    lateinit var navigatorHolder: NavigatorHolder
     private lateinit var binding: ActivityMainBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
             .also { setContentView(it.root) }
 
+        initViews()
+
         if (savedInstanceState == null) {
-            supportFragmentManager
-                .beginTransaction()
-                .add(fragmentContainer, PostScreenFragment())
-                .commit()
+            viewModel.currentTabState.observe(this) { renderTabState(it) }
+            viewModel.onFirstViewAttach()
+        }
+    }
+
+    override fun openTab(tabTag: TabTag) {
+        val fm = supportFragmentManager
+        var currentFragment: Fragment? = null
+        val fragments = fm.fragments
+        for (f in fragments) {
```

Comment:
давай без алфавитов, не есть хорошая практика
```suggestion
        for (fragment in fragments) {
```

Reply:
алфавит убрал, по нормальному назвал